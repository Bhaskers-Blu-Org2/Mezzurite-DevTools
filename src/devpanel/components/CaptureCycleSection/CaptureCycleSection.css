@media only screen {
  .capture-cycle--components {
    display: flex;
    flex-wrap: wrap;
    list-style-type: none;
    margin: 16px;
    padding: 0;
  }

  .capture-cycle--section-header {
    font-size: 20px;
    font-weight: 500;
    margin: 12px 16px 0;
    text-align: center;
  }

  .capture-cycle--section-sub-header {
    font-size: 16px;
    font-weight: 400;
    margin: 8px 16px 0;
    text-align: center;
  }
}

@media only screen and (min-width: 540px) {
  .capture-cycle--section-header,
  .capture-cycle--section-sub-header {
    text-align: left;
  }
}
