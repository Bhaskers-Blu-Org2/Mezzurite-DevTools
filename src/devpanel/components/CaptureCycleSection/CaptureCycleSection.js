import React from 'react';
import { arrayOf, node, number, shape, string } from 'prop-types';

import ComponentTiming from '../ComponentTiming/ComponentTiming';
import './CaptureCycleSection.css';

const CaptureCycleSection = (props) => (
  props != null && props.components != null && props.components.length > 0 && <section>
    {props.heading != null && <h3 className='capture-cycle--section-header'>{props.heading}</h3>}
    {props.subheading != null && <h4 className='capture-cycle--section-sub-header'>{props.subheading}</h4>}
    <ul className='capture-cycle--components'>
      {props.components.map((component, timingIndex) =>
        <ComponentTiming
          key={`capture-cycle-${props.captureCycleIndex}-component-${timingIndex}`}
          name={component.componentName}
          loadTime={component.componentLoadTime}
        />
      )}
    </ul>
  </section>
);

CaptureCycleSection.propTypes = {
  captureCycleIndex: number,
  components: arrayOf(shape({
    componentLoadTime: number,
    componentName: string
  })),
  heading: string,
  subheading: node
};

export default CaptureCycleSection;
