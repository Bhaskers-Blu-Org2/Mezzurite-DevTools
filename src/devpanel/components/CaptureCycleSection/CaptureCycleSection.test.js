import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import CaptureCycleSection from './CaptureCycleSection';

describe('CaptureCycleSection.js', () => {
  const renderer = new ShallowRenderer();

  it('should not render when the props are null', () => {
    const tree = renderer
      .render(<CaptureCycleSection />);
    expect(tree).toBeFalsy();
  });

  it('should not render when the components are null', () => {
    const tree = renderer
      .render(<CaptureCycleSection components={null} />);
    expect(tree).toBeFalsy();
  });

  it('should not render when the components are empty', () => {
    const tree = renderer
      .render(<CaptureCycleSection components={null} />);
    expect(tree).toBeFalsy();
  });

  it('should render the capture cycle section', () => {
    const tree = renderer
      .render(<CaptureCycleSection components={[
        {
          componentLoadTime: 14.2,
          componentName: 'componentName'
        }
      ]} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render the capture cycle section with a captureCycleIndex', () => {
    const tree = renderer
      .render(<CaptureCycleSection
        captureCycleIndex={4}
        components={[
          {
            componentLoadTime: 14.2,
            componentName: 'componentName'
          }
        ]}
      />);
    expect(tree).toMatchSnapshot();
  });

  it('should render the capture cycle section with a heading', () => {
    const tree = renderer
      .render(<CaptureCycleSection
        components={[
          {
            componentLoadTime: 14.2,
            componentName: 'componentName'
          }
        ]}
        heading='heading'
      />);
    expect(tree).toMatchSnapshot();
  });
});
