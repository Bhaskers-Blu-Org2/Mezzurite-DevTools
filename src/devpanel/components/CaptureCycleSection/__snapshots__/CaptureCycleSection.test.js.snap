// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaptureCycleSection.js should render the capture cycle section 1`] = `
<section>
  <ul
    className="capture-cycle--components"
  >
    <ComponentTiming
      loadTime={14.2}
      name="componentName"
    />
  </ul>
</section>
`;

exports[`CaptureCycleSection.js should render the capture cycle section with a captureCycleIndex 1`] = `
<section>
  <ul
    className="capture-cycle--components"
  >
    <ComponentTiming
      loadTime={14.2}
      name="componentName"
    />
  </ul>
</section>
`;

exports[`CaptureCycleSection.js should render the capture cycle section with a heading 1`] = `
<section>
  <h3
    className="capture-cycle--section-header"
  >
    heading
  </h3>
  <ul
    className="capture-cycle--components"
  >
    <ComponentTiming
      loadTime={14.2}
      name="componentName"
    />
  </ul>
</section>
`;

exports[`CaptureCycleSection.js should render the capture cycle section with a subheading 1`] = `
<section>
  <h4
    className="capture-cycle--section-sub-header"
  >
    <span>
      subheading
    </span>
  </h4>
  <ul
    className="capture-cycle--components"
  >
    <ComponentTiming
      loadTime={14.2}
      name="componentName"
    />
  </ul>
</section>
`;
